### ISSUE TITLE
WSL Nix installation db.sqlite is busy

### ISSUE THREAD

**Issue by grandynguyen** [NixOS/nix#2292]
On a fresh install of Ubuntu on WSL when I run:

`curl https://nixos.org/nix/install | sh`

I get:

```
 % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2472  100  2472    0     0   1892      0  0:00:01  0:00:01 --:--:--  1891
downloading Nix 2.0.4 binary tarball for x86_64-linux from 'https://nixos.org/releases/nix/nix-2.0.4/nix-2.0.4-x86_64-linux.tar.bz2' to '/tmp/nix-binary-tarball-unpack.eyxxE5LXqk'...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 21.0M  100 21.0M    0     0   898k      0  0:00:23  0:00:23 --:--:-- 1487k
performing a single-user installation of Nix...
directory /nix does not exist; creating it by running 'mkdir -m 0755 /nix && chown grandy /nix' using sudo
[sudo] password for grandy:
copying Nix to /nix/store...............................
initialising Nix database...
Nix: creating /home/grandy/.nix-profile
installing 'nix-2.0.4'
building '/nix/store/9v9rz64kdblx991qw8q7472yhisz0rs2-user-environment.drv'...
created 6 symlinks in user environment
killing process 247: No such process
unpacking channels...
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy (SQLITE_PROTOCOL)
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy (SQLITE_PROTOCOL)
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy (SQLITE_PROTOCOL)
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy (SQLITE_PROTOCOL)
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy (SQLITE_PROTOCOL)
warning: SQLite database '/nix/var/nix/db/db.sqlite' is busy (SQLITE_PROTOCOL)
```

I tried waiting 10 minutes. Will leave it running while I'm away.

**Comment by grandynguyen** [comment]
#1203 Seems to be the issue.

**Comment by DanielFabian** [comment]
I think this is a duplicate of #2651 now. Following the two workarounds (sqlite-wal and sandbox) got it to work on WSL for me today.

**Comment by razvan-flavius-panda** [comment]
I used the method mentioned by @rdbuf, and it appeared to it was stuck for a very long time on Unpacking, but in the end, it finished successfully.

**Comment by domenkozar** [comment]
This is fixed in master and 2.3.6
